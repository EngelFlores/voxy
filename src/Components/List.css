.list{
  overflow-x: auto;
}

.list table {
  border-collapse: collapse;
}

td, th {
  border: 1px solid #ddd;
}

tr th:last-child {
  cursor: pointer;
}
