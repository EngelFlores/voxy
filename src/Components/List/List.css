.list{
  overflow-x: auto;
}

.list table {
  border-collapse: collapse;
}

td, th {
  border: 1px solid rgba(102, 100, 100, 0.24);
}

