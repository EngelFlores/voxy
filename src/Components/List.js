import React, { useState, useEffect } from 'react';
import { Columns } from './Columns'
import { SearchBar } from './SeachBar'
import initialData from '../data/information';
import "./List.css"


const List = () => {
  const [data, setData] = useState([]);
  const [filteredList, setFilteredList] = useState([]);

  useEffect(() => {
    setData(initialData)
    setFilteredList(initialData)
  }, [])

  const mostStudied = () => {
    const sortedList = data.sort((firstElement, secondElement) => secondElement.hoursStudied - firstElement.hoursStudied)
    return setFilteredList([...sortedList])
  }

  const leastStudied = () => {
    const sortedList = data.sort((firstElement, secondElement) => firstElement.hoursStudied - secondElement.hoursStudied)
    return setFilteredList([...sortedList])
  }

  const filterList = (event) => {
    const search = event.target.value
    const filtered = data.filter(element => {
      return element.email.toLowerCase().indexOf(search) !== -1 || element.primaryGroup.toLowerCase().indexOf(search) !== -1
    })
    return setFilteredList(filtered)
  }

  return (
    <>
      <SearchBar onChange={filterList}></SearchBar>
      <button onClick={mostStudied}>Most Studied</button>
      <button onClick={leastStudied}>Least Studied</button>
      <div className="list">
        <table>
          <thead>
            <tr>
              <th>Email</th>
              <th>First Name</th>
              <th>Last Name</th>
              <th>Primary Group</th>
              <th>Phone Number</th>
              <th>Hours Studied</th>
            </tr>
          </thead>
          <tbody>
            {filteredList.map(information => {
              return <Columns key={information.id} information={information}></Columns>
            })}
          </tbody>
        </table>
      </div>
    </>
  )
}

export { List }
