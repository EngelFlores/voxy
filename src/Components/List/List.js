import React, { useState } from 'react';
import { Columns } from '../Columns/Columns'
import { SearchBar } from '../SeachBar'
import "./List.css"


const List = (props) => {
  const {data} = props
  const [filteredList, setFilteredList] = useState(data);

  const mostStudied = () => {
    const sortedList = data.sort((firstElement, secondElement) => secondElement.hoursStudied - firstElement.hoursStudied)
    return setFilteredList([...sortedList])
  }

  const leastStudied = () => {
    const sortedList = data.sort((firstElement, secondElement) => firstElement.hoursStudied - secondElement.hoursStudied)
    return setFilteredList([...sortedList])
  }

  const filterList = (event) => {
    const search = event.target.value
    const filtered = data.filter(element => {
      return element.email.toLowerCase().indexOf(search.toLowerCase()) !== -1 || element.primaryGroup.toLowerCase().indexOf(search.toLowerCase()) !== -1
    })
    return setFilteredList(filtered)
  }

  return (
    <>
      <SearchBar onChange={filterList}></SearchBar>
      <button data-testid="most-studied" onClick={mostStudied}>Most Studied</button>
      <button data-testid="least-studied" onClick={leastStudied}>Least Studied</button>
      <div className="list">
        <table>
          <thead>
            <tr>
              <th>Email</th>
              <th>First Name</th>
              <th>Last Name</th>
              <th>Primary Group</th>
              <th>Phone Number</th>
              <th>Hours Studied</th>
            </tr>
          </thead>
          <tbody data-testid="columns">
            {filteredList.map(information => {
              return <Columns key={information.id} information={information}></Columns>
            })}
          </tbody>
        </table>
      </div>
    </>
  )
}

export { List }
