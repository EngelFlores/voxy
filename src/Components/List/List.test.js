import { render, waitForElement, fireEvent } from '@testing-library/react';
import React from 'react'

import { List } from './List'

describe('Should render right information in the list', () => {
    const mock = [{
        "id": 53,
        "firstName": "Susanna",
        "lastName": "Legion",
        "email": "slegiona@prlog.org",
        "primaryGroup": "pretium",
        "hoursStudied": 2857,
        "phoneNumber": "152-210-9118"
    }, {
        "id": 352,
        "firstName": "Letta",
        "lastName": "Fryatt",
        "email": "lfryattb@furl.net",
        "primaryGroup": "imperdiet sapien",
        "hoursStudied": 1795,
        "phoneNumber": "543-217-4959"
    }]

    test('Render filtered information from user', async () => {
        const { getByTestId } = render(<List data={mock} />)
        const searchInput = await waitForElement(() => getByTestId('search-input'))
        const list = await waitForElement(() => getByTestId('columns'))
        const value = "lfryattb@furl.net"
        fireEvent.change(searchInput, { target: { value } })

        expect(list).toHaveTextContent('Letta')
    })

    test('Sort users have studied the most', async () => {
        const { getByTestId } = render(<List data={mock} />)
        const sortListButton = await waitForElement(() => getByTestId('most-studied'))
        const list = await waitForElement(() => getByTestId('columns'))
        fireEvent.click(sortListButton)
        
        expect(list).toMatchSnapshot()
    })

})