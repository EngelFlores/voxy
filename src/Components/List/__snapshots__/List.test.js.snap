// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render right information in the list Sort users have studied the most 1`] = `
<tbody
  data-testid="columns"
>
  <tr>
    <td
      data-testid="email"
    >
      slegiona@prlog.org
    </td>
    <td
      data-testid="first-name"
    >
      Susanna
    </td>
    <td
      data-testid="last-name"
    >
      Legion
    </td>
    <td
      data-testid="primary-group"
    >
      pretium
    </td>
    <td
      data-testid="phone-number"
    >
      152-210-9118
    </td>
    <td
      data-testid="hours-studied"
    >
      2857
    </td>
  </tr>
  <tr>
    <td
      data-testid="email"
    >
      lfryattb@furl.net
    </td>
    <td
      data-testid="first-name"
    >
      Letta
    </td>
    <td
      data-testid="last-name"
    >
      Fryatt
    </td>
    <td
      data-testid="primary-group"
    >
      imperdiet sapien
    </td>
    <td
      data-testid="phone-number"
    >
      543-217-4959
    </td>
    <td
      data-testid="hours-studied"
    >
      1795
    </td>
  </tr>
</tbody>
`;
